ocaml/typing/btype.ml

```diff
@@ -597,6 +597,25 @@ let label_name = function
   | Optional s
   | Position s -> s
 
+let optionally_apply_call_pos ~(arg : Parsetree.expression) ~application_loc =
+  let loc = { arg.pexp_loc with loc_ghost = true } in
+  let none_case =
+    Ast_helper.Exp.case
+      (Ast_helper.Pat.construct ~loc { loc; txt = Lident "None" } None)
+      (let loc = application_loc in
+       Ast_helper.Exp.extension ~loc ({ loc; txt = "src_pos" }, PStr []))
```

Comment:
It's a bit strange that we're using `src_pos` here. IIUC that's different than the `call_pos` we'd get from omitting the argument entirely.

Reply:
Oh sorry, I've realized this conversation is still open. I don't think this comment is still relevant.

The `scr_pos` there was constructed with location of the function application, so they would both have the same location as ommitting the argument entirely.